src/sbefw/core/sbeConsole.C

```diff
@@ -188,27 +190,21 @@ static void uartPutChar(char c)
         if(rc != SBE_SEC_OPERATION_SUCCESSFUL)
         {
             SBE_ERROR(SBE_FUNC " LSR read error.");
-            break;
-        }
-        if(data == LSR_BAD)
-        {
+        } else if(data == LSR_BAD) {
             SBE_ERROR(SBE_FUNC " LSR_BAD data error.");
-            break;
-        }
-        if(loops >= DELAY_LOOPS)
-        {
+        } else if(loops >= DELAY_LOOPS) {
             SBE_ERROR(SBE_FUNC " FIFO timeout.");
-            break;
-        }
-
-        rc = writeReg(THR, c);
-        if(rc != SBE_SEC_OPERATION_SUCCESSFUL)
-        {
-            SBE_ERROR(SBE_FUNC " failure to write THR");
-            break;
-        }
+        } else {
+	    tx_room = 16;
```

Comment:
Stewart, Once we enter into "if (tx_room < 1)" and get into any error [ LSR read error, LSR_BAD or FIFO timeout ] then "tx_room = 16" never executes. And Line number 206: "tx_room--" leads to wrong value. Is this case cross check ?